node/full_node_integration_test.go

```diff
@@ -321,36 +351,46 @@ func testSingleAggreatorSingleFullNodeSingleLightNode(t *testing.T) {
 	ds, _ := store.NewDefaultInMemoryKVStore()
 	_ = dalc.Init([8]byte{}, nil, ds, log.TestingLogger())
 	_ = dalc.Start()
-	sequencer, _ := createNode(aggCtx, 0, false, true, false, keys, &wg, t)
-	fullNode, _ := createNode(ctx, 1, false, false, false, keys, &wg, t)
+	sequencer, _ := createNode(aggCtx, 0, false, true, false, keys, t)
+	fullNode, _ := createNode(ctx, 1, false, false, false, keys, t)
 
 	sequencer.(*FullNode).dalc = dalc
 	sequencer.(*FullNode).blockManager.SetDALC(dalc)
 	fullNode.(*FullNode).dalc = dalc
 	fullNode.(*FullNode).blockManager.SetDALC(dalc)
 
-	lightNode, _ := createNode(ctx, 2, false, false, true, keys, &wg, t)
+	lightNode, _ := createNode(ctx, 2, false, false, true, keys, t)
 
 	require.NoError(sequencer.Start())
 	require.NoError(fullNode.Start())
 	require.NoError(lightNode.Start())
 
-	time.Sleep(3 * time.Second)
+	require.NoError(testutils.Retry(300, 100*time.Millisecond, func() error {
```

Comment:
since we have so many of these checks, i think we can even refactor the helper I wrote above to a generic helper.  We could have 2 helpers

```go
func verifyNodesSynced(node1, node2 *FullNode) {
	return testutils.Retry(300, 100*time.Millisecond, func() error {
		node1Height := node1.Store.Height()
		node2Height := node2.Store.Height()
		if node1Height == node2Height {
			return nil
		}
		return fmt.Errorf("nodes not synced: node1 at height %v, node2 at height %v", node1Height, node2Height)
	})
}

func verifyHeight(node *FullNode, h unit64) {
	return testutils.Retry(300, 100*time.Millisecond, func() error {
		nodeHeight := node.(*FullNode).Store.Height()
		if nodeHeight == h {
			return nil
		}
		return fmt.Errorf("node at height %v, expected height %v", nodeHeight, h)
	})
}
```